core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -104,8 +104,10 @@ default List<String> listSchemaNames(ConnectorSession session)
      * cannot be queried.
      * @see #getView(ConnectorSession, SchemaTableName)
      * @see #getMaterializedView(ConnectorSession, SchemaTableName)
+     * @deprecated Implement {@link #getTableHandle(ConnectorSession, SchemaTableName, Optional, Optional)}.
```

Comment:
It would be nice if the new method javadoc contained an explanation about startVersion/endVersion